api/src/damnit_api/filewatcher/utils.py

```diff
@@ -0,0 +1,18 @@
+from pathlib import Path
+
+import aiofiles
+
+from .. import get_logger
+
+logger = get_logger()
+
+
+async def fetch_file_data(file_path: Path, with_content: bool = False):
+
+    try:
+        async with aiofiles.open(file_path, encoding="utf-8") as f:
+            return await f.read()
+    except OSError as e:
+        logger.warning("Error reading %r, retrying… %s", file_path, e)
+
+    return ""
```

Comment:
The logging shouldn't say retrying if it isn't. :wink: Maybe it's also nicer to not catch the error - I assume that FastAPI will then give an HTTP 500 error, so the frontend knows that something has gone wrong, rather than getting an empty context file (of course, how you handle the error on the frontend is another question).

And then it's probably not worth this being in its own module, either.